smart-contracts/contracts/mixins/MixinKeys.sol

```diff
@@ -210,6 +218,31 @@ contract MixinKeys is
     return _ownerOf[_tokenId];
   }
 
+  /**
+  * @notice Update transfer and cancel rights for a given key
+  * @param _tokenId The id of the key to assign rights for
+  * @param _keyManager The address with the manager's rights for the given key.
+  * Setting _keyManager to address(0) means the keyOwner is also the keyManager
+  * */
+  //
+  function setKeyManagerOf(
+    uint _tokenId,
+    address _keyManager
+  ) public
+    isKey(_tokenId)
+  {
+    require(
+      keyManagerOf[_tokenId] == msg.sender ||
+      isLockManager(msg.sender) ||
+      keyManagerOf[_tokenId] == address(0), 'UNAUTHORIZED_KEY_MANAGER_UPDATE'
```

Comment:
Can merge this with the next require with 
``` ... ||
(keyManagerOf[_tokenId] == address(0) && isKeyOwner(...)), 'UNAUTH...
```

Reply:
yeah, that works for me. will change it.